Port ExampleSign to 1.4.4

xNarnia opened the pull request:
### What are the changes to ExampleMod?
**ExampleSign**
This example was previously apart of 1.3. I've updated it to compile on 1.4.4.
Enclosed is a working example on how to create a modded sign. 

**There are examples of how to:**
- Use the Main.sign array
- Define the tile to be a sign
- Getting and setting the text of the sign 

**Changes from ExampleSign 1.3 to 1.4.4:**
- Port to 1.4.4
- Reduce clutter in example
- Adjust anchors to be much simpler

### Do these changes need additional documentation?
No, the ExampleSign is thorough enough to get the point across.

xNarnia committed:
Port ExampleSign to 1.4.4

- Reduce clutter in example
- Adjust anchors to be much simpler
- Port to 1.4.4

direwolf420 commented on ExampleMod/Content/Items/ExampleSignItem.cs:
```diff
@@ -0,0 +1,37 @@
+{
+	public class ExampleSignItem : ModItem
+	{
+		public override void SetStaticDefaults()
+		{
+			Item.DefaultToPlaceableTile(ModContent.TileType<ExampleSign>(), 0);
+		}
+
+		public override void SetDefaults()
+		{
+			Item.width = 26;
+			Item.height = 22;
+			Item.maxStack = 99;
+			Item.useTurn = true;
+			Item.autoReuse = true;
+			Item.useAnimation = 15;
+			Item.useTime = 10;
+			Item.useStyle = ItemUseStyleID.Swing;
+			Item.consumable = true;
+			Item.value = 500;
+			Item.createTile = ModContent.TileType<ExampleSign>();
+		}
```
The `DefaultToPlaceableTile` call is misplaced. Move it to `SetDefaults`, and then you can get rid of everything there except `Item.value`, `Item.width`, and `Item.height`. See other tiles in ExampleMod for reference.

xNarnia replied:
Updated with https://github.com/tModLoader/tModLoader/pull/4269/commits/b3429f2eb320a9b6257ae914a3d3010ec38a88a3

direwolf420 commented on ExampleMod/Content/Tiles/ExampleSign.cs:
```diff
@@ -0,0 +1,76 @@
+		public override void SetStaticDefaults()
+		{
+			// Note: The max amount of signs in a world is 1000 (the size of the tileSign array).
+			Main.tileSign[Type] = true;
```
Are you saying that if there are 1000 tiles loaded, and the last one happens to be ExampleSign, it will IOOB?

xNarnia replied:
This comment has an error, thank you for catching it! The comment is mistakenly stating tileSign when it should be stating the Main.sign[] max length is 1000. 
I've updated the comment to be more clear, and moved it to the PlaceInWorld section where it makes more sense.

Updated with https://github.com/tModLoader/tModLoader/pull/4269/commits/b3429f2eb320a9b6257ae914a3d3010ec38a88a3

direwolf420 commented on ExampleMod/Content/Tiles/ExampleSign.cs:
```diff
@@ -0,0 +1,76 @@
+            // The ID is the array index for our new Sign in the Main.sign[] array.
+            int signId = Sign.ReadSign(i, j, true);
+
+            Main.sign[signId].text = "I am a sign, right-click me!";
```
Localization? See the Localization guide and ExampleMod usages of LocalizedText

xNarnia replied:
Localization added with https://github.com/tModLoader/tModLoader/pull/4269/commits/b3429f2eb320a9b6257ae914a3d3010ec38a88a3

direwolf420 commented on ExampleMod/Content/Tiles/ExampleSign.cs:
```diff
@@ -0,0 +1,76 @@
+			int signId = Sign.ReadSign(i, j);
+			string signText = Main.sign[signId].text;
+            
+			ChatHelper.BroadcastChatMessage(NetworkText.FromLiteral(signText), Color.White);
```
~~In the same vein as above, make it use `LocalizedText.ToNetworkText`~~ nvm, sign text can be customized so not feasible

direwolf420 commented on ExampleMod/Content/Tiles/ExampleSign.cs:
```diff
@@ -0,0 +1,76 @@
+            return true;
+        }
+    }	
+}
```
Missing end-of-file newline

xNarnia replied:
Updated with https://github.com/tModLoader/tModLoader/pull/4269/commits/b3429f2eb320a9b6257ae914a3d3010ec38a88a3

xNarnia committed:
Localization, fix item defaults, minor fixes

- Added localization support
- Fixed comment error mistakenly stating tileSign instead of Main.sign[]
- Added eof newline
- Update ExampleSignItem defaults to trim unnecessary values

JavidPack committed:
Fix formatting and bugs in ExampleSign

JavidPack committed:
Document sign related fields and sets

JavidPack commented:
Thanks a bunch.
